streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -590,6 +590,30 @@ public void prepareRecycle() {
         log.info("Closed and recycled state");
     }
 
+    @Override
+    public void preparePoll() {
```

Comment:
I could not find dedicated test for this method. Could you please add one?

Reply:
Name: good idea. I wanted to use this as a "bag of things to do before/after" the poll, as we do with prepareCommit / postCommit, but you are right that it's not necessarily a good pattern, and we only have one thing in the bag, so let's use a specific name.

Reply:
As for testing: I think it only makes sense to test this in conjunction with pausing, unless we want to leak `PartitionGroup`. At first, I tried to refactor the code so that `PartitionGroup` is injected, but after reconsideration, I think we should only test the externally visible behavior here and leave `PartitionGroup` encapsulated. So then, I think the `shouldPauseAndResumeBasedOnBufferedRecords` is actually a good test for this already.